cmd/performance-profile-creator/cmd/root.go

```diff
@@ -160,17 +165,33 @@ func getProfileData(args profileCreatorArgs) (*ProfileData, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get reserved and isolated CPUs for %s: %v", nodeName, err)
 	}
+	kernelArgs := getAdditionalKernelArgs(args.powerConsumptionMode)
 	profileData := &ProfileData{
 		reservedCPUs:           reservedCPUs,
 		isolatedCPUs:           isolatedCPUs,
 		nodeSelector:           mcp.Spec.NodeSelector,
 		performanceProfileName: args.profileName,
 		topologyPoilcy:         args.tmPolicy,
 		rtKernel:               args.rtKernel,
+		additionalKernelArgs:   kernelArgs,
 	}
 	return profileData, nil
 }
 
+func getAdditionalKernelArgs(powerMode string) []string {
+	var kernelArgs []string
+	switch powerMode {
+	case validPowerConsumptionModes[0]:
+		kernelArgs = []string{}
+	case validPowerConsumptionModes[1]:
+		kernelArgs = additionalKernelArgs[:3]
+	case validPowerConsumptionModes[2]:
+		kernelArgs = additionalKernelArgs
```

Comment:
not a fan of this approach. I prefer to have different sets and to merge them as needed. Also, the kernel args (among other things) are data we treat as code, so it should be inside some pkg, albeit possibly a tiny one.

Reply:
I felt this was much cleaner than joining multiple sets but okay to update if that's the way we want to implement it. Regarding the second part, I am not sure I understand completely what you mean by data being treated as code. You mean that we should be defining additionalKernelArgs in profilecreator pkg or want me to create a separate pkg just for this?

Reply:
Had a discussion with @fromanirh about this and got the clarification on why it should be implemented as a set. Implementing as a slice and using the indices to obtain a portion of the slice has risk in the case the master slice expands or shrinks causing the boundary to change leading to unexpected results.

Also got the clarification that its best to move getAdditionalKernelArgs to a pkg.